### ISSUE TITLE
CMS Editors Should Not Be Able to Rename/Re-URL-alias Vet Center Facility Service Pages

### ISSUE THREAD

**Issue by stefaniefgray** [department-of-veterans-affairs/va.gov-cms#11367]
## Describe the defect
All Vet Centers have various facility services that are necessary to add to each Vet Center homepage. Many of these services are listed as required, while some are optional.

Vet Center facility services include "Community engagement", "Grief and bereavement counseling", "Mental health care", and more. They are listed within Drupal as the content type "[Vet Center - Facility Service](https://prod.cms.va.gov/admin/content?title=&type=vet_center_facility_health_servi&moderation_state=All&owner=All)"

Please go to the "Services" section of the Cleveland Vet Center homepage to see how they are listed: ([production](https://prod.cms.va.gov/cleveland-vet-center/service/cleveland-vet-center-grief-and-bereavement-counseling), [live site](https://www.va.gov/cleveland-vet-center/))

To see how Services are added to a Vet Center homepage when in Editing mode: https://staging.cms.va.gov/node/3642/edit

## Problem Statement:

It is currently possible for Editors to edit Vet Center facility service nodes and change which service each one is linked to. 

This also changes the title and URL for the service, removes it from the associated Vet Center homepage, and makes it difficult to track down later on without knowing the node ID.

## Why is this a problem?

Because the Vet Center team keeps running into facility service issues during moderation that are holding up the publication of various Vet Centers.

#### Example 1:
The Vet Center moderation team observed that the Tampa Vet Center was missing the **required** Vet Center facility services "Couples and Family Counseling" and "Suicide Prevention." These nodes were thought to have disappeared, and could not be located at all within the CMS.

- The "Couples and Family Counseling" node was actually re-named to "Tampa Vet Center - Telehealth," and the URL was changed to [/tampa-vet-center/service/tampa-vet-center-telehealth](https://prod.cms.va.gov/tampa-vet-center/service/tampa-vet-center-telehealth).
- Here is proof that this was originally the "Couples and Family Counseling" page, not Telehealth. This was changed by the Editor on 10/5/2022: https://prod.cms.va.gov/node/40279/revisions/592629/view
- The same thing happened regarding the original Tampa Vet Center Suicide Prevention service page, which became "Homeless Veteran Care," and the URL was changed to [/tampa-vet-center/service/tampa-vet-center-homeless-veteran-care](https://prod.cms.va.gov/tampa-vet-center/service/tampa-vet-center-homeless-veteran-care)
- Proof from revision history that this node was also modified by the Editor on 10/5/2022: https://prod.cms.va.gov/node/41436/revisions/594958/view

#### Example 2:
The Vet Center moderation team noted that the Cleveland Vet Center was missing the **required** Vet Center facility service "Veteran Connections"

- This service page did exist under this title in the past, but had been renamed to "Cleveland Vet Center - Telehealth" and the URL had been changed to[ /cleveland-vet-center/service/cleveland-vet-center-telehealth](https://prod.cms.va.gov/cleveland-vet-center/service/cleveland-vet-center-telehealth)
- Proof from revision history that the Editor had re-assigned this page to Telehealth on 8/18/2022: https://prod.cms.va.gov/node/41577/revisions

****

## To Reproduce
1. Log into staging.cms.va.gov as a Vet Center **Content Editor** (can ask @stefaniefgray for login info)
2. Go to `/admin/content` and narrow it down to content type "Vet Center - Facility Service"
3. Find an existing Vet Center Facility Service that is associated with your assigned Vet Center homepage
4. Observe the **title,** **service name,** and **URL** of this Vet Center Facility Service before clicking "Edit"
5. Use the **Service name** dropdown to select a different health service than the one currently listed and click "Save." 
6. You will see that the **title,** **service name,** and **URL** of this Vet Center Facility Service has changed.

(Note: If you cannot save the page after Step 5, choose a different health service from the dropdown menu. The site will not you create duplicates of ones that already exist for that Vet Center.)

## AC / Expected behavior
- [ ] Editors should not be able to change the title, service name, and/or URL of existing Vet Center facility service codes

#### URL `tampa-vet-center/service/tampa-vet-center-addiction-and-substance-abuse-care`

![Screen Shot 2022-10-27 at 4 02 00 PM](https://user-images.githubusercontent.com/9059428/198406727-89e0a356-239f-4fed-9545-a1a55d17c251.png)

### Changing which service is listed on this node as a Vet Center Content Editor:
![Screen Shot 2022-10-27 at 4 04 20 PM](https://user-images.githubusercontent.com/9059428/198406974-8f96fee6-71d5-4617-9222-10302a0f248b.png)

#### New URL `/tampa-vet-center/service/tampa-vet-center-lgbtq-veteran-care`

![Screen Shot 2022-10-27 at 4 06 18 PM](https://user-images.githubusercontent.com/9059428/198407110-ce091982-800c-4cfb-bab5-c8ec9473e94b.png)

 ## Additional context
DSVA Slack thread re: Tampa Vet Center: https://dsva.slack.com/archives/CDHBKAL9W/p1666893981346209

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [x] `⭐️ Facilities`
- [x] `⭐️ User support`

**Comment by dsinla** [comment]
Follow up with @stefaniefgray - is it correct to say that a Required Service was omitted as a result of this unintentional rename?
If so, we can put some guardrails on checking for the existence of required service

**Comment by stefaniefgray** [comment]
@dsinla All Vet Center required services were added to the live site properly when the CMS was built.

HOWEVER, there are no guardrails to keep editors from changing an _existing_ required service page to being linked to a different service altogether. 

**Example:**
- The Cleveland Vet Center was missing the required Vet Center facility service "Veteran Connections"
- This required service page actually did exist as "Cleveland Vet Center - Veteran Connections" up until August 2022, but had been renamed to "Cleveland Vet Center - Telehealth."
- Moreover, the URL had been changed to[ /cleveland-vet-center/service/cleveland-vet-center-telehealth](https://prod.cms.va.gov/cleveland-vet-center/service/cleveland-vet-center-telehealth) -- so, without some Drupal magic, there's no way of knowing that the Veteran Connections page had existed at this node in the past.

**Editors should not be allowed to modify Vet Center service page titles and URLs!** This has led to some wonky Vet Center service counts, i.e. Michelle Middaugh finding 299 service pages for "Veteran Connections" instead of 300.

Proof from revision history that the Editor had re-assigned this page to Telehealth on 8/18/2022: https://prod.cms.va.gov/node/41577/revisions

**Comment by dsinla** [comment]
Proposed: If this is a required service, the field for selecting the service is disabled for non-admins.

**Comment by stefaniefgray** [comment]
Michelle Middaugh mentioned today that the Lincoln Vet Center changed their “Recreation and creative arts therapy” health service to “Workshops and classes”: https://prod.cms.va.gov/node/50420/revisions

See thread: https://dsva.slack.com/archives/C01K37HRUAH/p1669918055184319

**Comment by omahane** [comment]
@stefaniefgray Confirmed that this is possible with the VAMC Facility Health Service names.

**Comment by davidmpickett** [comment]
Renamed and retagged the issue based on the fact that this affects Vet Centers and VAMCs. @Dottisea I'm flagging this as a good candidate for refinement. The problem is very well documented (kudos @stefaniefgray!) and affects almost every CMS editor. 

I think there's also potential that there could be a solution to this issue that also addresses #12232. Basically any type of state-change of a Facility Service node causes problems, so there should be guardrails in place to prevent editors AND admins from doing that, because the correct current behavior is delete & recreate per #12182

**Comment by JayDarnell** [comment]
I have retitled this and updated the ACs to reflect this discussion: https://dsva.slack.com/archives/C0FQSS30V/p1675722212286809

Basically we want to make it so a Vet Center Facility Service, once created, cannot be changed to a different type of service (which would ultimately change the title and the URL). Allowing this to happen makes for a confusing experience in terms of tracking changes over time and can be extremely confusing for editors and admins.
